app/src/main/kotlin/no/nav/arbeidsgiver/notifikasjon/skedulert_påminnelse/SkedulertPåminnelseRepository.kt

```diff
@@ -0,0 +1,57 @@
+package no.nav.arbeidsgiver.notifikasjon.skedulert_påminnelse
+
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import no.nav.arbeidsgiver.notifikasjon.hendelse.HendelseModel
+import java.time.Instant
+import java.time.LocalDate
+import java.util.*
+
+class SkedulertPåminnelseRepository {
+    private val mutex = Mutex()
+    private val indexedLookup = HashMap<UUID, SkedulertPåminnelse>()
+    private val påminnelseQueue = TreeMap<Instant, MutableList<SkedulertPåminnelse>>()
```

Comment:
Nøklene her er `Instant`, mens i utgått så er det `LocalDate`. Det vil være mange flere entries i dette map-et enn i utgått sitt. Kanskje går det fint? Eller så kunne man hatt mer grov-kornet lookup, og så gjort linært søk for å finne aktuelle. Men kanskje overkill om det ikke er noen ytelsesproblem.

Reply:
1e073c1c45e9231936fb6a6cc3b1a1fe9e289a41